### PR TITLE
SONAR-7680 Fallback to class name when Sensor name is not provided

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

File: sonar-scanner-engine/src/main/java/org/sonar/batch/sensor/SensorWrapper.java [hunk 1]
```diff
@@ -59,6 +59,10 @@ public void analyse(Project module, org.sonar.api.batch.SensorContext context) {
 
   @Override
   public String toString() {
-    return descriptor.name() + (LOG.isDebugEnabled() ? " (wrapped)" : "");
+    if (descriptor.name() != null) {
+      return descriptor.name() + (LOG.isDebugEnabled() ? " (wrapped)" : "");
```
Comment (henryju) [inline sonar-scanner-engine/src/main/java/org/sonar/batch/sensor/SensorWrapper.java]: Can you please remove this "wrapped" addition? Not sure this is useful now that new Sensor API is out of beta.

File: sonar-scanner-engine/src/main/java/org/sonar/batch/util/BatchUtils.java [hunk 2]
```diff
@@ -61,7 +61,10 @@ public static String encodeForUrl(@Nullable String url) {
   public static String describe(Object o) {
     try {
       if (o.getClass().getMethod("toString").getDeclaringClass() != Object.class) {
-        return o.toString();
+        String str = o.toString();
+        if(str != null) {
```
Comment (henryju) [inline sonar-scanner-engine/src/main/java/org/sonar/batch/util/BatchUtils.java]: Code style

**Comment by henryju** [comment]
Few comments but LGTM

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [SensorWrapper.java#L31](https://github.com/SonarSource/sonarqube/blob/46fc4d356197006558704ac7e32c8b1631981e29/sonar-scanner-engine/src/main/java/org/sonar/batch/sensor/SensorWrapper.java#L31): Remove this unused "LOG" private field. [rule](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1068)

**Pull request merged**